packages/remix-cloudflare-workers/worker.ts

```diff
@@ -121,20 +127,28 @@ export function createEventHandler({
     return response;
   };
 
-  return (event: FetchEvent) => {
+  return async (event: FetchEvent | Evt) => {
+    const respondWith = (payload: Response) => {
+      if ('respondWith' in event && typeof event.respondWith === 'function') {
+        event.respondWith(payload);
+        return;
+      }
+      return payload;
```

Comment:
The mixed return here is _tolerable_ (since Wrangler/Miniflare will run without typechecking), but it does mean that the fetch handler no longer conforms to `ExportedHandlerFetchHandler` from `@cloudflare/workers-types`. The code is similar enough that it would be annoying to split it into 2 separate functions, but I gave it a crack and couldn’t figure out how to use conditional types in a way that resolved it properly.